### PR TITLE
Removed program enrollment from dialog

### LINKED ISSUE #1050: Console warnings on home page
I got some console warnings related to the localStorage work we did recently. However I haven't actually noticed anything not working due to them

screenshot from 2016-09-15 13-29-47

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Part of #1113
Fixes #1050 

#### What's this PR do?
Removes the code to enroll the user at signup. Instead there is just a login button in the dialog.

#### How should this be manually tested?
- You should not be enrolled in any program after logging in via the dialog box.
- You should not see any console warnings anymore

#### Any background context you want to provide?
@aliceriot let me know if there's other code that should be cut, I tried to be conservative here since I wasn't familiar with the recent changes here

#### Screenshots (if appropriate)
screenshot from 2016-09-26 16-34-21

### CODE REVIEW

**Comment by aliceriot** [comment]
@noisecapella sure I'll take a look!

**Comment by aliceriot** [comment]
some additional things that can / should be deleted (might be helpful to look at https://github.com/mitodl/micromasters/pull/992)
- code in `cms/models.py` to stick programs into `SETTINGS`, on both the program page and the home page
- tests in `actions/signup_test.js`
- tests in `reducers/signup_test.js`
- there's code in App.js which looks for programs added on the homepage and, if found, adds an enrollment
- two selectors in `signup_dialog.scss` (`.program-select` and `.terms-of-service-text`) which are not found anywhere any longer

that's all I can think of right now...

**Comment by noisecapella** [comment]
Ok I think the tests should be passing now and the stuff you listed has been removed

**Comment by aliceriot** [comment]
cool, taking a look!

**Comment by aliceriot** [comment]
functionality looks good, reading code now

File: static/js/containers/SignupDialog.js [hunk 1]
```diff
@@ -65,44 +37,20 @@ const SignupDialog = ({
...
-    dispatch(setProgram(value));
-  };
-
-  return {
-    setDialogVisibility: visible => dispatch(setDialogVisibility(visible)),
-    setProgram: setProgramHelper,
-  };
-};
+const mapDispatchToProps = dispatch => ({
+  setDialogVisibility: visible => dispatch(setDialogVisibility(visible)),
```
Comment (aliceriot) [inline static/js/containers/SignupDialog.js]: I think this could be changed to use the `createSimpleActionHelper` function (in `util/redux.js`)

Comment (aliceriot) [inline static/js/containers/SignupDialog.js]: dunno why I didn't use that before! haha

File: static/js/reducers/signup_dialog.js [hunk 2]
```diff
@@ -1,24 +1,19 @@
...
 
 export const INITIAL_SIGNUP_STATE = {
   dialogVisibility: false,
-  program: null,
 };
 
 export const signupDialog = (state: SignupState = INITIAL_SIGNUP_STATE, action) => {
   switch (action.type) {
   case SET_DIALOG_VISIBILITY:
     return { ...state, dialogVisibility: action.payload };
```
Comment (aliceriot) [inline static/js/reducers/signup_dialog.js]: since there's only one key now, do we even need to return `...state`?

Comment (aliceriot) [inline static/js/reducers/signup_dialog.js]: probs doesn't really matter

Comment (noisecapella) [inline static/js/reducers/signup_dialog.js]: Yeah, I'd rather leave it just in case we need to add one later

**Comment by noisecapella** [comment]
Ok, comments addressed

**Comment by noisecapella** [comment]
I think it should be fixed now

**Comment by noisecapella** [comment]
Nevermind... two more tests to fix

**Comment by noisecapella** [comment]
Ok, tests are passing

**Comment by aliceriot** [comment]
awesome, taking a look!

File: static/js/containers/ProfileFormContainer.js [hunk 3]
```diff
@@ -155,15 +155,16 @@ class ProfileFormContainer extends React.Component {
...
       errors: errors,
       ui: ui,
       updateProfile: this.updateProfile.bind(this, isEdit),
       saveProfile: this.saveProfile.bind(this, isEdit),
       fetchProfile: this.fetchProfile,
-    }, ...this.simpleActionHelpers(), ...this.asyncActionHelpers()
-    );
+      ...this.simpleActionHelpers(),
+      ...this.asyncActionHelpers()
+    };
```
Comment (aliceriot) [inline static/js/containers/ProfileFormContainer.js]: much cleaner! :100:

**Pull request merged**